buildSrc/src/main/kotlin/org/sonarsource/kotlin/buildsrc/tasks/FetchRuleMetadata.kt

```diff
@@ -0,0 +1,44 @@
+package org.sonarsource.kotlin.buildsrc.tasks
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.TaskAction
+import org.gradle.kotlin.dsl.dependencies
+import org.gradle.kotlin.dsl.provideDelegate
+import org.gradle.kotlin.dsl.repositories
+import org.gradle.process.ExecResult
+
+abstract class FetchRuleMetadata : DefaultTask() {
+
+    @get:Input
+    val ruleKey: String by project
+
+    @TaskAction
+    fun downloadMetadata(): ExecResult {
+        project.repositories {
+            maven {
+                url = project.uri("https://repox.jfrog.io/repox/sonarsource-private-releases")
+                authentication {
+                    credentials {
+                        val artifactoryUsername: String by project
+                        val artifactoryPassword: String by project
+                        username = artifactoryUsername
+                        password = artifactoryPassword
+                    }
+                }
+            }
+        }
+
+        val ruleApi = project.configurations.create("ruleApi")
+        project.dependencies {
+            ruleApi("com.sonarsource.rule-api:rule-api:2.0.0.1885")
+        }
+
+        return project.javaexec {
+            classpath = project.files(ruleApi.resolve())
```

Comment:
AFAIK rule-api requires `GITHUB_TOKEN` env variable to be set. Do you think it will be nice to add such check before downloading and running rule-api?

Reply:
Interesting! I don't think I have it configured and I can't find where the api tool uses it either. I'll find out more about this. If it is indeed needed, I think you may have a good point here!

Reply:
Maybe this was changed, I remember it was the case previously

Reply:
Yes, it was changed by [this](https://github.com/SonarSource/sonar-rule-api/commit/39f0a55f89b17fb5c65119449e9d3f19dfc75ea6#diff-d9f85f5dfe7cae4b4117d496f199a4b18f022605437d9cdd38f50bf2f853eb73) commit